posthog/queries/session_query.py

```diff
@@ -28,6 +30,10 @@ def __init__(
         self._filter = filter
         self._team = team
         self._session_id_alias = session_id_alias
+        # the session_replay_events table couldn't have existed before this date in any instance of PostHog
+        self._can_use_session_replay_events = self._filter.date_from and self._filter.date_from > parse(
+            "2023-05-10T00:00:00Z"
+        )
```

Comment:
I think we would more generally need to have a way of checking "does this team actually have meaningful replay data?". Otherwise we are assuming that every account records replay data all the time... Would be easy enough to just have a cached query that checks for replay data within the requested period (maybe matching up the number of distinct `$session_id` between events and replay_events. 

That way we only use the optimised query _if_ we are pretty certain that there is the same or more session data contained in replay_events

Reply:
Yeah, I was wondering about a single-run query into a global or something... I don't think you can reasonably be running a version of PostHog that contains this check, are ingesting events with $session_id and aren't ingesting into session_replay_events though?

I guess the only downside to a query that this team has any session_replay_events is the impact on all the `FuzzyInt` assertions I'd have to change 🤣